lib/screens/authentication/recover/recover_account_success/interactor/viewmodels/recover_account_success_bloc.dart

```diff
@@ -2,16 +2,43 @@ import 'dart:async';
 
 import 'package:bloc/bloc.dart';
 import 'package:equatable/equatable.dart';
+import 'package:hashed/domain-shared/base_use_case.dart';
 import 'package:hashed/domain-shared/page_command.dart';
 import 'package:hashed/domain-shared/page_state.dart';
+import 'package:hashed/screens/authentication/recover/recover_account_success/usecase/fetch_recover_account_success_data.dart';
 
 part 'recover_account_success_event.dart';
 part 'recover_account_success_state.dart';
 
 class RecoverAccountSuccessBloc extends Bloc<RecoverAccountSuccessEvent, RecoverAccountSuccessState> {
   RecoverAccountSuccessBloc(String userAccount) : super(RecoverAccountSuccessState.initial(userAccount)) {
     on<FetchInitialData>(_fetchInitialData);
+    on<OnRefreshTapped>(_onRefreshTapped);
+    on<OnRecoverFundsTapped>(_onRecoverFundsTapped);
   }
 
-  Future<void> _fetchInitialData(FetchInitialData event, Emitter<RecoverAccountSuccessState> emit) async {}
+  Future<void> _fetchInitialData(FetchInitialData event, Emitter<RecoverAccountSuccessState> emit) async {
+    final Result<ResultData> result = await FetchRecoverAccountSuccessData().run(state.userAccount);
+    emit(state.copyWith(pageState: PageState.loading));
+
+    if (result.isValue) {
+      final data = result.asValue!.value;
+      emit(state.copyWith(
+        pageState: PageState.success,
+        recoveredAccount: data.recoveredAccount,
+        recoverAmount: data.amountToRecover,
+      ));
+    } else {
+      emit(state.copyWith(pageState: PageState.failure));
+    }
+  }
+
+  FutureOr<void> _onRefreshTapped(OnRefreshTapped event, Emitter<RecoverAccountSuccessState> emit) {
+    add(const FetchInitialData());
+  }
+
+  FutureOr<void> _onRecoverFundsTapped(OnRecoverFundsTapped event, Emitter<RecoverAccountSuccessState> emit) {
+    /// Here we recover the users funds?
+    // TODO(Nik): Recover funds calls here.
```

Comment:
@n13  aI left a series of TODOs for you. 
They are all around making calls to repos and parsing the data into a format the UI understands.

Reply:
Yeah cool this all makes sense I will get on this and finish the screens with it. Merging for now